client/server/pages/analytics.js

```diff
@@ -18,14 +19,21 @@ const logAnalyticsThrottled = throttle( function ( sectionName, duration, target
  * Only logs if the request context contains a `sectionName` attribute.
  */
 export function logSectionResponse( req, res, next ) {
-	const startRenderTime = new Date();
+	const startRenderTime = process.hrtime.bigint();
 
-	res.on( 'finish', function () {
-		const context = req.context || {};
-		if ( context.sectionName ) {
-			const duration = new Date() - startRenderTime;
-			logAnalyticsThrottled( context.sectionName, duration, context.target );
+	res.on( 'close', () => {
+		if ( ! req.context?.sectionName ) {
+			return;
 		}
+		const { user, sectionName, target, usedSSSRHandler } = req.context;
+
+		const statKey = `${ sectionName }.loggedin_${ !! user }.ssr_${ usedSSSRHandler }`;
+
+		const responseTime = Number(
```

Comment:
this is copy/pasted from `middleware/logging`, which does basically the exact same thing, except it puts the "duration" in the Calypso server logs, which are imported to logstash.

Reply:
The statsd logging expects timings in milliseconds, and for that, `new Date()` is sufficient as it returns values with ms resolution. `process.hrtime` measures time in nanoseconds and we don't need such big precision.

Reply:
Switched back to the old approach, but swapped it for `Date.now()`